### PR TITLE
Add docs on managing change

### PR DESCRIPTION
**Pull request by 36degrees**
Split out the documentation on deprecations from the versioning documentation, as the new documentation supercedes it.

### CODE REVIEW

File: docs/contributing/managing-change.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Managing change
+
+When making changes to the codebase that could affect our users, we want to minimise the number of breaking changes in a major version that cannot be made before updating to the major version.
```
Comment (vanitabarrett) [inline docs/contributing/managing-change.md:3]: Slightly unsure what we mean by the second half of this sentence, "...in a major version that cannot be made before updating to the major version". Are we talking about minimising the number of breaking changes that a user might be unable to make in their own service/port?

File: docs/contributing/managing-change.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+///
+/// @deprecated Use govuk-multiply(number, 2) instead.
+///   See https://github.com/alphagov/govuk-frontend/issues/1234
+@function govuk-double($number) {
+  @warn "govuk-double($number) is deprecated. Use govuk-multiply($number, 2) instead.";
+  @return govuk-multiply($number, 2);
+}
+```
+
+### Deprecating an parameter for a Sass mixin or function
```
Comment (vanitabarrett) [inline docs/contributing/managing-change.md:73]:
```suggestion
### Deprecating a parameter for a Sass mixin or function
```

File: docs/contributing/versioning.md [hunk 3]
```diff
@@ -41,32 +41,6 @@ This is to ensure the community can get involved with the decision.
 
 Make an active effort to involve the community, this might be in the form of presentations or meetings.
 
-## Deprecation
```
Comment (vanitabarrett) [inline docs/contributing/versioning.md:44]: There is an anchor link to this section [further up in the file](https://github.com/alphagov/govuk-frontend/pull/2679/files#diff-86a4fd213ed466a2707fbf35c759f4bf8fdf9a24139864757994ef8bc22dcdb1R34) which will break now the section has been removed.

File: docs/contributing/versioning.md [hunk 4]
```diff
@@ -41,32 +41,6 @@ This is to ensure the community can get involved with the decision.
...
-
-### Sass doc deprecation annotations
-
-When deprecating Sass, you can use the [deprecated annotation](http://sassdoc.com/annotations/#deprecated).
-
-If there is an alias, for example if you are renaming something, you can use the [alias annotation](http://sassdoc.com/annotations/#alias).
-
-See [an example of deprecating a Sass mixin on GitHub](https://github.com/alphagov/govuk-frontend/blob/9424d87ed54764d2d8afe35d6e0077ee43d231e1/src/helpers/_grid.scss#L20-L26).
-
 ## Migration
```
Comment (vanitabarrett) [inline docs/contributing/versioning.md:44]: Should this be part of the new doc too?

**Review by vanitabarrett — CHANGES_REQUESTED** [review]

File: docs/contributing/managing-change.md (hunk 1 above)
Comment (36degrees) [reply docs/contributing/managing-change.md:3]: We want as many breaking changes as possible to be 'actionable' in minor releases that come prior to the major release.

So if a user has followed the release notes by the time they upgrade to the next major version there should be few changes they actually have to make at that point.

Does that make sense? Agree it's clumsily worded at the minute – maybe @claireashworth can help?

File: docs/contributing/versioning.md (hunk 3 above)
Comment (36degrees) [reply docs/contributing/versioning.md:44]: I've resolved this by removing the migration section (which didn't really add anything over what was in the managing change doc) and updating the link to point to the managing change doc instead.

File: docs/contributing/managing-change.md (hunk 2 above)
Comment (36degrees) [reply docs/contributing/managing-change.md]: Good catch!

File: docs/contributing/versioning.md (hunk 4 above)
Comment (36degrees) [reply docs/contributing/versioning.md]: I don't think there's anything in there that's not said in the new doc in more detail, so have removed.

File: docs/contributing/managing-change.md (hunk 1 above)
Comment (claireashworth) [reply docs/contributing/managing-change.md:3]: How about something like this:

Breaking changes should, wherever possible, be actionable by the users. By doing this the users will have fewer actions to do to complete a major release.

Comment (vanitabarrett) [reply docs/contributing/managing-change.md:3]: Ah I see, thanks for explaining. So it's more about, where possible, telling users about changes we're making in patch/feature releases so they're given a heads up earlier and are given the opportunity to make changes to their service in advance to prepare?

Comment (36degrees) [reply docs/contributing/managing-change.md:3]: Exactly that 👍🏻

Comment (36degrees) [reply docs/contributing/managing-change.md:3]: @claireashworth I think the missing bit is that the changes should ideally be actionable in prior minor or patch versions.

That's not always going to be possible but we should aim to do it where it makes sense. The rest of the doc mostly provides strategies for doing so.

Comment (vanitabarrett) [reply docs/contributing/managing-change.md:3]: Could we say something like:

```Where possible, users should be notified about upcoming breaking changes in a patch or feature release. By doing this, users have the opportunity to make the relevant changes in advance, so they have fewer actions to complete when we do a major release.```

Comment (36degrees) [reply docs/contributing/managing-change.md]: I've pushed a change that tries to simplify the 'intro' down to the bare minimum, and let the other sections go in to more detail on how it helps.

Thoughts?

Comment (claireashworth) [reply docs/contributing/managing-change.md]: The intro looks good. A couple of style guide things - change 'whilst' to 'while' and expand 'don't' (negative verbs are written in full for gov.uk style)

**Review by vanitabarrett — APPROVED** [review]

**Pull request merged**